code_generation/code_gen.py

```diff
@@ -117,17 +117,45 @@ def render_dataset_class_maps(self, template_path: Path, data_path: Path, output
 
         self.render_template(template_path=template_path, output_path=output_path, all_map=all_map)
 
+    def render_metadata_types(self, template_path: Path, data_path: Path, output_path: Path):
```

Comment:
Mainly I wanted to separate responsibilities a bit (C++ from python generated code). Also, it kind of provided it but within `render_dataset_class_maps`, so I just made its own implementation.

Reply:
i believe most of the things are the same right? it's ok to have multiple things defined that you don't actually use in the jinja template, but since we don't do language-specific case-sensitive stuff, i don't really see a reason to duplicate this function.

the separation of concerns should be restricted to the actual template. only if we start generating different conventions for Python, R, Java, C#, ... whatever other languages we might support in the future, then we might want to start making separate generators for each of those languages.

but for the time being, it's almost only data declarations, so it's fine to combine them

Reply:
Changes made. You were right, I misunderstood the your comment initially.